crates/l2/proposer/prover_server.rs

```diff
@@ -351,68 +360,57 @@ impl ProverServer {
-        let mut attempts = 0;
-        let max_retries = 100;
-        let retry_secs = std::time::Duration::from_secs(5);
-        while attempts < max_retries {
-            match self
-                .send_proof(block_number, &seal, image_id, journal_digest)
-                .await
-            {
-                Ok(tx_hash) => {
-                    info!(
-                        "Sent proof for block {block_number}, with transaction hash {tx_hash:#x}"
-                    );
-                    break; // Exit the while loop
-                }
+        // The `verify` function in the `OnChainProposer` contract has two `require` checks:
+        //      "OnChainProposer: block not committed"
+        //      "OnChainProposer: block already verified"
+        //
+        // We should never encounter the "block not committed" error, as it's handled
+        // by the check: `last_committed_block < block_number`.
+        //
+        // If we get the "block already verified" error, it means we are submitting a `block_number`
+        // smaller than the `lastVerifiedBlock` tracked by the contract, which is stored in `ProverServer::last_verified_block`.
+        // We shouldn't encounter this error either.
+        //
+        // Both errors will trigger a panic, along with any errors resulting from sending the transactions.
+        match self
+            .send_proof(block_number, &seal, image_id, journal_digest)
+            .await
+        {
+            Ok(tx_hash) => {
+                info!("Sent proof for block {block_number}, with transaction hash {tx_hash:#x}");
+            }
 
-                Err(e) => {
-                    warn!("Failed to send proof to block {block_number:#x}. Error: {e}");
-                    let eth_client_error = format!("{e}");
-                    if eth_client_error.contains("block not committed") {
-                        attempts += 1;
-                        if attempts < max_retries {
-                            warn!("Retrying... Attempt {}/{}", attempts, max_retries);
-                            sleep(retry_secs).await; // Wait before retrying
-                        } else {
-                            error!("Max retries reached. Giving up on sending proof for block {block_number:#x}.");
-                            panic!("Failed to send proof after {} attempts.", max_retries);
-                        }
-                    } else {
-                        error!("Failed to send proof to block {block_number:#x}. Manual intervention required: {e}");
-                        panic!("Failed to send proof to block {block_number:#x}. Manual intervention required: {e}");
-                    }
-                }
+            Err(e) => {
+                error!("Failed to send proof to block {block_number:#x}. Manual intervention required: {e}");
+                panic!("Failed to send proof to block {block_number:#x}. Manual intervention required: {e}");
```

Comment:
Instead of panicking here, let's just backoff, wait a while and try sending the transaction again. An error when sending the proof (or on any other request to the L1 for that matter) is something that can (and will) happen every now and then, everywhere where we interact with L1 we should follow a pattern like this; retry after a little while, not panic.